### PR TITLE
[W5.8b][T11-A4]Eka Buyung Lienadi

### PR DESCRIPTION
**Pull request by yungyung04**

### CODE REVIEW

File: doc/DeveloperGuide.md [hunk 1]
```diff
@@ -94,6 +94,10 @@ Use case ends.
 2. Should be able to hold up to 1000 persons.
 3. Should come with automated unit tests and open source code.
 4. Should favor DOS style commands over Unix-style commands.
+5. Should never crashes during usage
+6. Should have specialized interfaces for [different devices] (#different devices)
+7. Developer guide and User guide should be updated for each new version and written concisely.
+8. Address book data should be integrated with a cloud storage to prevent undesirable data loss.
```
Comment (yungyung04) [inline doc/DeveloperGuide.md]: if point 8 suits more to be a functional requirement, how should I rephrase it?

File: doc/DeveloperGuide.md [hunk 2]
```diff
@@ -94,6 +94,10 @@ Use case ends.
 2. Should be able to hold up to 1000 persons.
 3. Should come with automated unit tests and open source code.
 4. Should favor DOS style commands over Unix-style commands.
+5. Should never crashes during usage
+6. Should have specialized interfaces for [different devices] (#different devices)
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md]: NFR should be specific, so better describe the `specialized interfaces`.

File: doc/DeveloperGuide.md [hunk 3]
```diff
@@ -104,3 +108,11 @@ Use case ends.
 ##### Private contact detail
 
 > A contact detail that is not meant to be shared with others
+
+##### Crashes
```
Comment (yungyung04) [inline doc/DeveloperGuide.md:112]: is this necessary?

File: doc/DeveloperGuide.md [hunk 4]
```diff
@@ -94,6 +94,10 @@ Use case ends.
 2. Should be able to hold up to 1000 persons.
 3. Should come with automated unit tests and open source code.
 4. Should favor DOS style commands over Unix-style commands.
+5. Should never [crashes] (#crashes) during usage
+6. Should have specialized interfaces for [different devices] (#different devices)
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md:98]: NFR should be specific or tangible, so you better describe more about the `specialized interface`.

File: doc/DeveloperGuide.md (hunk 3 above)
Comment (jeffryhartanto) [reply doc/DeveloperGuide.md:112]: Not necessary, better to put the explanation on the NFR itself.

File: doc/DeveloperGuide.md [hunk 5]
```diff
@@ -94,6 +94,10 @@ Use case ends.
...
+7. Developer guide and User guide should be updated for each new version and written concisely.
+8. Address book data should be integrated with a cloud storage to prevent undesirable data loss.
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md:100]: This is more toward non-NFR, so you can remove it.

File: doc/DeveloperGuide.md [hunk 6]
```diff
@@ -94,6 +94,10 @@ Use case ends.
 2. Should be able to hold up to 1000 persons.
 3. Should come with automated unit tests and open source code.
 4. Should favor DOS style commands over Unix-style commands.
+5. Should never [crashes] (#crashes) during usage
+6. Should have specialized interfaces for [different devices] (#different devices)
+7. Developer guide and User guide should be updated for each new version and written concisely.
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md:99]: `concisely` is not a specific or tangible word (subjective).

File: doc/DeveloperGuide.md [hunk 7]
```diff
@@ -104,3 +108,11 @@ Use case ends.
...
 
 > A contact detail that is not meant to be shared with others
+
+##### Crashes
+
+> program stops execution unnormally without showing an error message.
+
+##### Different devices
+
+> PC, smartphones, tablets.
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md:118]: You can also put this on the NFR itself.

**Comment by jeffryhartanto** [comment]
@yungyung04 
Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**